### DISCUSSION TITLE
Call Objeck from C++

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#131] [General]
Is it possible? It's possible to call Java from C++ using JNI:

https://www.baeldung.com/jni#2-using-objects-and-calling-java-methods-from-native-code

**Comment by objeck** [comment]
Yes, however, it is less high-level than Java. 

Such code includes a ``foo.obl`` and ``foo.(so,dll)`` set of files for deployment. The Objeck interface is portable, while the share libraries are platform dependent. 

Here is a SDL2 example:

- [Objeck interface](https://github.com/objeck/objeck-lang/blob/master/core/compiler/lib_src/sdl2.obs#L1009)
- [C++ call](https://github.com/objeck/objeck-lang/blob/master/core/lib/sdl/sdl.cpp#L1673)

**Reply by ghost** [reply]
I don't think this is the right example. You should have a look at the JNI article I posted above. The method is implemented in Java and they call it from C++. Your example doesn't call any method written in Objeck from C++. It's normal wrapping of C++ to call C++ from Objeck.

**Reply by ghost** [reply]
JNI is not only about calling C/C++ from Java but it can also call Java from C/C++ :smile:
